csharp/autobuilder/Semmle.Autobuild.Shared/BuildActions.cs

```diff
@@ -216,12 +217,18 @@ XmlDocument IBuildActions.LoadXml(string filename)
 
         public string EnvironmentExpandEnvironmentVariables(string s) => Environment.ExpandEnvironmentVariables(s);
 
-        public void DownloadFile(string address, string fileName)
+        static async Task DownloadFileAsync(string address, string filename)
         {
-            using var webClient = new WebClient();
-            webClient.DownloadFile(address, fileName);
+            using var httpClient = new HttpClient();
+            using var request = new HttpRequestMessage(HttpMethod.Get, address);
+            using var contentStream = await (await httpClient.SendAsync(request)).Content.ReadAsStreamAsync();
+            using var stream = new FileStream(filename, FileMode.Create, FileAccess.Write, FileShare.None, 4096, true);
+            await contentStream.CopyToAsync(stream);
         }
 
+        public void DownloadFile(string address, string fileName) =>
+            DownloadFileAsync(address, fileName).Wait();
```

Comment:
I'm not sure if this could cause a deadlock or not. Ideally we should be async all the way up, but I just realized that this is our first async call in the application.

For exception handling it looks like `.GetAwaiter().GetResult()` is a bit better. https://jaliyaudagedara.blogspot.com/2019/06/taskwait-vs-taskgetawaitergetresult.html
But `Exception` is caught, so it probably doesn't matter.